packages/ubuntu_bootstrap/lib/installer/installer_wizard.dart

```diff
@@ -68,76 +59,37 @@ class _InstallWizard extends ConsumerWidget {
-      if (welcome)
-        Routes.welcome: WizardRoute(
-          builder: (_) => const WelcomePage(),
-          userData: WizardRouteData(step: InstallationStep.locale.index),
-          onLoad: (_) => WelcomePage.load(context, ref),
-        ),
-      Routes.rst: WizardRoute(
-        builder: (_) => const RstPage(),
-        onLoad: (_) => RstPage.load(ref),
-      ),
-      Routes.keyboard: WizardRoute(
-        builder: (_) => const KeyboardPage(),
-        userData: WizardRouteData(step: InstallationStep.keyboard.index),
-        onLoad: (settings) => KeyboardPage.load(ref),
-      ),
-      Routes.network: WizardRoute(
-        builder: (_) => const NetworkPage(),
-        userData: WizardRouteData(step: InstallationStep.network.index),
-        onLoad: (_) => NetworkPage.load(ref),
-      ),
-      Routes.refresh: WizardRoute(
-        builder: (_) => const RefreshPage(),
-        userData: WizardRouteData(step: InstallationStep.refresh.index),
-        onLoad: (_) => RefreshPage.load(ref),
-      ),
-      Routes.source: WizardRoute(
-        builder: (_) => const SourceWizard(),
-        userData: WizardRouteData(step: InstallationStep.source.index),
-        onLoad: (_) => SourcePage.load(ref),
-      ),
-      Routes.secureBoot: WizardRoute(
-        builder: (_) => const SecureBootPage(),
-        userData: WizardRouteData(step: InstallationStep.type.index),
-        onLoad: (_) => SecureBootPage.load(ref),
-      ),
-      Routes.storage: WizardRoute(
-        builder: (_) => const StorageWizard(),
-        userData: WizardRouteData(step: InstallationStep.storage.index),
-        onLoad: (_) => StorageWizard.load(ref),
-      ),
+    final pageConfig = ref.watch(pageConfigProvider);
+    final preInstallRoutes = <String, WizardRoute>{
+      for (final pageName in pageConfig.includedPages)
+        Routes.routeMap[pageName]!:
+            InstallationStep.fromName(pageName)!.toRoute(context, ref)
     };
 
     return WizardBuilder(
       initialRoute: Routes.initial,
-      userData: WizardData(totalSteps: InstallationStep.values.length),
-      routes: <String, WizardRoute>{
+      userData: WizardData(totalSteps: preInstallRoutes.length),
+      routes: {
         Routes.loading: WizardRoute(
           builder: (_) => const LoadingPage(),
+          userData: const WizardRouteData(
+            hasPrevious: false,
+            hasNext: false,
+          ),
           onReplace: (_) => LoadingPage.init(context, ref).then((_) => null),
         ),
-        ...preInstall,
-        Routes.confirm: WizardRoute(
-          builder: (_) => const ConfirmPage(),
-          userData: WizardRouteData(step: InstallationStep.storage.index),
-          onLoad: (_) => ConfirmPage.load(ref),
-        ),
+        ...preInstallRoutes,
         Routes.install: WizardRoute(
           builder: (_) => const InstallPage(),
-          onLoad: (_) => InstallPage.load(context, ref),
+          onLoad: (_) => const InstallPage().load(context, ref),
```

Comment:
Now that I'm seeing those two lines together - shouldn't we keep the `load` method static?

Reply:
Then we can't have the pages conform to an interface since static methods can't be overridden and then all pages need to be defined one by one, like it was before. Since they are const classes it is not any performance degradation here either, even though that wouldn't really have mattered.